pkg/clustermanager/cluster_manager.go

```diff
@@ -689,6 +694,56 @@ func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster,
 	return nil
 }
 
+// UpgradeClusterAPI incorporates EKS-A cluster status changes for cluster upgrade process.
+func (c *ClusterManager) UpgradeClusterAPI(ctx context.Context, managementCluster, workloadCluster *types.Cluster, newClusterSpec *cluster.Spec, provider providers.Provider) error {
```

Comment:
Could we implement this outside of cluster manager?
I would like to get rid of it eventually, and this new code is gonna stay for a long time.